### PR TITLE
Config backend updates

### PR DESCRIPTION
**Pull request by bamandel**

### CODE REVIEW

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            return createErrorResponse(HttpMethod.PUT, e);
+        }
+        if (responseObject != null) {
+            return Response.ok(responseObject).status(Response.Status.BAD_REQUEST).build();
+        }
+        return Response.noContent().build();
+    }
+
+    private Response createErrorResponse(final HttpMethod httpMethod, final Throwable e) {
+        final String msg = String.format("Exception performing {} : {}", httpMethod.name(), e.getMessage());
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java]: Does this format work for `String.format(...)`? I thought it was only for log messages.

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+                final List<String> allowedMethods = project.getAllowedMethods();
+                if (allowedMethods != null && !allowedMethods.isEmpty() && allowedMethods.contains("GET") && allowedMethods.contains("PUT")) {
+                    blackDuckProjects.add(project.getName());
+                }
+            }
+        }
+        return blackDuckProjects;
+    }
+
+    private void setBlackDuckPolicyRules(final BlackDuckServicesFactory blackDuckServicesFactory, final BlackDuckJiraConfigSerializable config) {
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java:295]: We might want to abstract a few parts of this method based on its size and complexity.

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/action/IssueCreationConfigActions.java [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+    }
+
+    public BlackDuckJiraConfigSerializable getCreatorCandidates() {
+        final BlackDuckJiraConfigSerializable config = new BlackDuckJiraConfigSerializable();
+        config.setCreatorCandidates(new TreeSet());
+
+        final SortedSet<String> creatorCandidates = getIssueCreatorCandidates();
+        config.setCreatorCandidates(creatorCandidates);
+
+        if (creatorCandidates.size() == 0) {
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/action/IssueCreationConfigActions.java]: Could we use `.isEmpty()`?

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/action/IssueCreationConfigActions.java [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        return config;
+    }
+
+    public BlackDuckJiraConfigSerializable getJiraProjects() {
+        final List<JiraProject> jiraProjects = getJiraProjects(projectManager.getProjectObjects());
+
+        final BlackDuckJiraConfigSerializable txProjectsConfig = new BlackDuckJiraConfigSerializable();
+        txProjectsConfig.setJiraProjects(jiraProjects);
+
+        if (jiraProjects.size() == 0) {
```
Comment (gkillough) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/action/IssueCreationConfigActions.java]: `.isEmpty()`?

**Review by gkillough — APPROVED** [review]

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java (hunk 1 above)
Comment (jamesrichard91) [reply src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java]: For String.format(...) you need to use %s for variables

Comment (bamandel) [reply src/main/java/com/blackducksoftware/integration/jira/config/controller/AccessConfigController.java]: Ya, my bad there. Already changed locally.

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+package com.blackducksoftware.integration.jira.config.controller.action;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.lang.StringUtils;
```
Comment (jamesrichard91) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java]: This should be commons lang3

File: src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+            return newConfig;
+        }
+    }
+
+    public BlackDuckJiraConfigSerializable getBlackDuckPolicies() {
+        final PluginSettingsWrapper pluginSettingsWrapper = createPluginSettingsWrapper();
+        final String policyRulesJson = pluginSettingsWrapper.getStringValue(PluginConfigKeys.BLACKDUCK_CONFIG_JIRA_POLICY_RULES_JSON);
+        final BlackDuckJiraConfigSerializable txConfig = new BlackDuckJiraConfigSerializable();
+
+        if (org.apache.commons.lang3.StringUtils.isNotBlank(policyRulesJson)) {
```
Comment (jamesrichard91) [inline src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java]: StringUtils should already be imported, we shouldn't need the full package here

**Review by jamesrichard91 — CHANGES_REQUESTED** [review]
Address the changes in the comments. The rest looks good!

**Pull request merged**